index.bs

```diff
@@ -782,7 +782,8 @@ WorkerNavigator includes NavigatorML;
 <script type=idl>
 enum MLDeviceType {
   "cpu",
-  "gpu"
+  "gpu",
+  "npu"
 };
```

Comment:
I wonder if it makes sense to swap the order, also depicting a possible default fallback path according to [comment](https://github.com/webmachinelearning/webnn/issues/623#issuecomment-2165827072) (eventually).

Reply:
If you mean ...

```
enum MLDeviceType {
  "cpu",
  "npu",
  "gpu"
};
```

...I'm *okay* with reordering them, but this enum just specifies the primary preferred device with no connection to fallback. If we want to add more control over fallback later, per one of the other options [listed here](https://github.com/webmachinelearning/webnn/issues/623#issuecomment-2063954107), that would be a separate attribute.

These devices have differing levels of generality:

- CPU - most general (can handle anything you throw it at, albeit not as quickly as others)
- GPU - less general (cannot handle branchy operators well or operators that dynamically allocate output memory, but tis very good with trivially parallelizable data)
- NPU - least general (more fixed function pipelines, less memory than either GPU or CPU...)

And different performance characteristics:

- CPU - slowest throughput
- NPU - faster throughput than CPU for parallelizable operators (especially convolution/GEMM) and less battery usage than the equivalent CPU or GPU execution
- GPU - fastest throughput but higher power usage, and usage can interfere with other GPU tasks like rendering

And different fallback characteristics (where fallback means you cannot execute all tasks on the given preferred device because of limitations like operator support or data type support):

- CPU - since the CPU is the most generic and can handle everything thrown at it, there is no need to "fallback" down to another device.
- GPU - unsupported operators or data types may fall back to CPU. "falling back" from GPU to NPU doesn't generally make sense because a GPU is a more generic superset to an NPU anyway, and if you've already ramped up the GPU for execution, then you might as well keep execution on it and avoid memory copies between devices.
- NPU - fallback to either GPU or CPU can make sense, especially if the NPU is on the same die and shares memory with either of those.

So if you want the enum order to correspond to fallback (from most generic to least generic), then the current order of {cpu, gpu, npu} makes sense.

Reply:
@fdwr thanks, makes sense!

Reply:
Minor point but: enum definition order in the WebIDL is not web-observable, so this is an "editorial" concern only, and there's not going to be any web-compat issue with reordering later. 

I would avoid relying on the IDL definition implying anything to either implementers or web developers. So adding prose explanation SGTM if we want to. But also, as noted in the last telecon I'm not sure we'll want to keep this API beyond the prototyping phase, so I also don't think we need to spend much time on it. But also also, for the same reason I think it's fine to add text and not agonize about making it perfect.